src/placeos-driver/driver_model.cr

```diff
@@ -22,16 +22,32 @@ struct PlaceOS::Driver::DriverModel
     include JSON::Serializable
 
     def initialize(
-      @functions = {} of String => Hash(String, Array(JSON::Any)),
+      @interface : Hash(String, Hash(String, JSON::Any))? = nil,
       @implements = [] of String,
       @requirements = {} of String => Array(String),
       @security = {} of String => Array(String),
       @settings = {type: "object", properties: {} of String => JSON::Any, required: [] of String}
     )
+      @interface ||= {} of String => Hash(String, JSON::Any)
+      @functions = nil
+    end
+
+    @[Deprecated("Use `#interface` instead of functions")]
+    def initialize(
+      @functions : Hash(String, Hash(String, Array(JSON::Any)))? = nil,
+      @implements = [] of String,
+      @requirements = {} of String => Array(String),
+      @security = {} of String => Array(String),
+      @settings = {type: "object", properties: {} of String => JSON::Any, required: [] of String}
+    )
+      @functions ||= {} of String => Hash(String, Array(JSON::Any))
+      @interface = nil
     end
 
     # Functions available on module, map of function name to args
-    property functions : Hash(String, Hash(String, Array(JSON::Any)))
+    property interface : Hash(String, Hash(String, JSON::Any))?
+    property functions : Hash(String, Hash(String, Array(JSON::Any)))?
```

Comment:
Both of these look to have seperate accessor methods that override this (L62 and L82)

Reply:
yes, temporarily as the migration occurs